lib/versioning/semver-coerced/index.ts

```diff
@@ -42,30 +50,40 @@ function getSatisfyingVersion(
   versions: string[],
   range: string
 ): string | null {
-  const coercedVersions = versions.map((version) => {
-    const coercedVersion = semver.coerce(version);
-    return coercedVersion ? coercedVersion.version : null;
-  });
+  const coercedVersions = versions
+    .map((version) => {
+      const coercedVersion = semver.coerce(version);
+      return coercedVersion ? coercedVersion.version : '';
+    })
+    .filter((version) => version !== '');
```

Comment:
```suggestion
  const coercedVersions = versions
    .map((version) => semver.coerce(version))
    .filter(Boolean);
```

Reply:
Applied this:-
```js
const coercedVersions = versions
    .map((version) => {
      const coercedVersion = semver.coerce(version);
      return coercedVersion ? coercedVersion.version : '';
    })
    .filter(Boolean);
    ```
    
    Using` .map((version) => semver.coerce(version))` results in null-check error.